### PR TITLE
fix building version 3.31 by using autoconf/libtools

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Review by beckermr — COMMENTED** [review]
See the PR for 3.8X. We should match that.

**Comment by mwittgen** [comment]
> See the PR for 3.8X. We should match that.

This PR has some weird build recipe as well like patching the config.stat with sed and creating the shared lib from the static libs.
Unfortunately the whole upstream package build is not consistent and varies from version to version.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,34 +1,14 @@
...
-    sed -i 's/-O3/-O3 -fPIC/g' config.status
-    
-fi
+export CPATH="${PREFIX}/include"
 
-./config.status
+./configure --prefix=$PREFIX --disable-dependency-tracking
 
 make -j ${CPU_COUNT}
+make install
```
Comment (beckermr) [inline recipe/build.sh:13]: I don't see this:

```
if [ "$(uname)" == "Darwin" ]; then

    ${CXX} -fPIC -shared -o $PREFIX/lib/libhealpix_cxx.dylib -Wl,-force_load $PREFIX/lib/libhealpix_cxx.a -Wl,-force_load $PREFIX/lib/libsharp.a -lomp -lcfitsio ${CXXFLAGS} ${LDFLAGS}

else

    ${CXX} -fPIC -shared -o $PREFIX/lib/libhealpix_cxx.so -Wl,--whole-archive $PREFIX/lib/libsharp.a $PREFIX/lib/libhealpix_cxx.a -Wl,--no-whole-archive ${CXXFLAGS} ${LDFLAGS}

fi
```

**Comment by mwittgen** [comment]
The result of the new build recipe for 3.31 are shared and static libs. Just deleting 
the static libs after relinking to shared in 3.80 is a slightly different product.
I am not sure how healpix_cxx is actually used by the conda-forge community, but the 
current package is of limited usability due to the missing shlib. Really playing it safe could is the build 
a new variant of 3.31 and have the users pick to use the old 3.31 or the new variant.

**Review by beckermr — CHANGES_REQUESTED** [review]
The static libs should never have been in this package in the first place. See our policy on this in the CFEP repo. People using the static libs from conda-forge are relying on an unsupported and brittle feature. 

I don't use this package and am only here as a core dev trying to help keep this package consistent with the rest of the community. 

There are two issues:

1. We need to separate the static libs into another package or delete them. 
2. We cannot vendor shared libs from other libs here. We attempted to fix this in the 3.80 build.

**Comment by mwittgen** [comment]
OK. This can be fixed. But this would break any current usage of healpix_cxx.
There is no other lib currently in the package

> The static libs should never have been in this package in the first place. See our policy on this in the CFEP repo. People using the static libs from conda-forge are relying on an unsupported and brittle feature.
> 
> I don't use this package and am only here as a core dev trying to help keep this package consistent with the rest of the community.
> 
> There are two issues:
> 
> 1. We need to separate the static libs into another package or delete them.
Easy fix, adding a delete
> 2. We cannot vendor shared libs from other libs here. We attempted to fix this in the 3.80 build.
The package would only vendor libhealpix_cxx.

**Comment by beckermr** [comment]
> OK. This can be fixed. But this would break any current usage of healpix_cxx.
> There is no other lib currently in the package

Breakage is fine here. We are free to break unsupported behavior.

**Comment by mwittgen** [comment]
Probably moving forward to 3.80 is a good idea after fixing 3.31

**Comment by beckermr** [comment]
If you have specific downstream code that builds against these shared libs, testing it against the libs in the 3.80 or this PR would be great. We are trying to bundle some stuff together and that might not be working how we thing it is.

**Comment by mwittgen** [comment]
Simplified the recipe. Only the shlib version is built, all test programs are dynamically linked.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,14 +10,20 @@ source:
   fn: Healpix_{{ version }}_2016Aug26.tar.gz
   url: https://downloads.sourceforge.net/project/healpix/Healpix_{{ version }}/Healpix_{{ version }}_2016Aug26.tar.gz
   sha256: {{ sha256 }}
+  patches:
+    - patches/Makefile.am.patch
 
 build:
```
Comment (beckermr) [inline recipe/meta.yaml:16]: We need a run export here.

Comment (beckermr) [inline recipe/meta.yaml:16]:
```suggestion
build:
  run_exports:
    - {{ pin_subpackage("healpix_cxx", max_pin="x.x") }}
```

**Review by beckermr — APPROVED** [review]

**Pull request merged**